.cirrus.yml

```diff
@@ -0,0 +1,78 @@
+_pipeline: &pipeline
+  info_script: |
+    meson --version
+    ninja --version
+  submodule-update_script: git submodule update --init --recursive
+  configure_script: meson build
+  compile_script: ninja -C build
+  test_script: ninja -C build test
+
+Debian (gcc)_task:
+  container:
+    image: snaipe/ci-meson:debian-10
+  setup_script: |-
+    apt-get update
+    apt-get install -y pkg-config cmake libgit2-dev libffi-dev libnanomsg-dev
+    pip3 install cram==0.7
+  <<: *pipeline
+
+Alpine (gcc,x86_64)_task:
+  container:
+    image: snaipe/ci-meson:alpine
+  setup_script: &alpine-deps |-
+    apk add --no-cache cmake libgit2-dev libffi-dev
+    pip3 install cram==0.7
+  <<: *pipeline
+
+Alpine (gcc,i386)_task:
+  container:
+    image: snaipe/ci-meson:alpine-x86
+  setup_script: *alpine-deps
+  <<: *pipeline
+
+Alpine (gcc,arm)_task:
+  container:
+    image: snaipe/ci-meson:alpine-armhf
+  env:
+    CIRRUS_SHELL: /sbin/qemu-sh
+  setup_script: *alpine-deps
+  <<: *pipeline
+
+Alpine (gcc,aarch64)_task:
+  container:
+    image: snaipe/ci-meson:alpine-aarch64
+  env:
+    CIRRUS_SHELL: /sbin/qemu-sh
+  setup_script: *alpine-deps
+  <<: *pipeline
+
+MacOS_task:
+  osx_instance:
+    image: mojave-xcode-10.2
+  env:
+    PKG_CONFIG_PATH: ${PKG_CONFIG_PATH}:/usr/local/opt/libffi/lib/pkgconfig
+  setup_script: |-
+    brew install meson cmake pkg-config libffi libgit2
+    pip3 install cram==0.7
+  <<: *pipeline
+
+FreeBSD_task:
+  freebsd_instance:
+    image_family: freebsd-12-0
+  setup_script: |-
+    pkg install -y py36-pip meson git libffi libgit2
+    pip-3.6 install cram==0.7
```

Comment:
empirically speaking) In FreeBSD, maintainers mercilessly deprecate and remove the packages from ports quite often than other packaging managements. Would probably make sense to keep pip version-less, or use the one from python [e.g.](https://github.com/Snaipe/Criterion/pull/287/files#diff-0f7067c31caedf7df2739c9b07efa122R31):

```diff
- pkg install -y py36-pip meson git libffi libgit2
- pip-3.6 install cram==0.7
+ pkg install -y cmake python meson git libffi libgit2
+ python -m ensurepip && python -m pip install --upgrade pip
+ python -m pip install --user cram==0.7
```

Reply:
Do they change these kind of versions & package names within a major FreeBSD release? (as in, is using freebsd-12-0 ensuring that py36-pip is there?)

In any case, I think I'll take your word for it here, especially if I want to easily upgrade images in the future.

Reply:
Yup, CI-like use-cases are not very FreeBSD-friendly. For example, I was pulling in a specific version of node.js in CI job, and suddenly (like hours later) someone just ninja'd in a newer version and "removed" the old one. Full story: https://github.com/cbsd/cbsd/issues/367#issuecomment-459356238. I ended up [scraping their webpage](https://github.com/sass/node-sass/blob/0aa9518068ed969300dfb3e6925449bbbff49c00/.cirrus.yml#L43-L45) for "whatever available version". 😅

ps - there are just too many package management systems, all with different/contradicting philosophies